src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2074,6 +2082,25 @@ namespace winrt::TerminalApp::implementation
         {
             _actionDispatch->DoAction(action);
         }
+
+        // After handling all the actions, then re-check the tabIndex. We might
+        // have been called as a part of a tab drag/drop. In that case, the
+        // tabIndex is actually relevant, and we need to move the tab we just
```

Comment:
I'm guessing this is the code to blame when I drag tab 1 out of a window and it becomes tab 1 in a new window...

Reply:
Oh, guess not. Why does it always become tab 1 when i drop it on a new window?